src/Translator/src/FileDumper/AssetMapperTranslationsFileDumper.php

```diff
@@ -0,0 +1,100 @@
+ */
+
+namespace Symfony\UX\Translator\FileDumper;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Translation\MessageCatalogueInterface;
+
+/**
+ * @author Maelan Le Borgne <maelan.leborgne@gmail.com>
+ *
+ * @final
+ *
+ * @experimental
+ *
+ * @phpstan-type Domain string
+ * @phpstan-type Locale string
+ * @phpstan-type MessageId string
+ */
+class AssetMapperTranslationsFileDumper implements TranslationFileDumperInterface
+{
+    private array $excludedDomains = [];
+    private array $domains = [];
+
+    public function __construct(
+        private string $dumpDir,
+        private Filesystem $filesystem,
+    ) {
+    }
+
+    public function dump(MessageCatalogueInterface ...$catalogues): void
+    {
+        $this->filesystem->mkdir($this->dumpDir);
+        $this->filesystem->remove($this->dumpDir.'/domains');
+        $this->filesystem->mkdir($this->dumpDir.'/domains');
+
+        $domainSplit = [];
+        foreach ($this->getTranslations(...$catalogues) as $translationId => $translationDetails) {
+            $domain = $translationDetails['domain'];
+            $locale = $translationDetails['locale'];
+            $domainFileName = str_ends_with($domain, MessageCatalogueInterface::INTL_DOMAIN_SUFFIX)
+                ? substr($domain, 0, -\strlen(MessageCatalogueInterface::INTL_DOMAIN_SUFFIX))
+                : $domain;
+
+            $domainSplit[$domainFileName] ??= [];
+            $domainSplit[$domainFileName][$translationId] ??= ['id' => $translationId, 'translations' => []];
+            $domainSplit[$domainFileName][$translationId]['translations'][$domain] ??= [];
+            $domainSplit[$domainFileName][$translationId]['translations'][$domain][$locale] = $translationDetails['message'];
+        }
+
+        foreach ($domainSplit as $fileName => $translations) {
+            $this->filesystem->dumpFile($this->dumpDir.'/domains/'.$fileName.'.js', sprintf(
+                "export default %s;\n",
+                json_encode($translations, \JSON_THROW_ON_ERROR | \JSON_PRETTY_PRINT)
+            ));
+        }
+    }
+
+    public function setExcludedDomains(array $domains = []): void
+    {
+        $this->excludedDomains = $domains;
+    }
+
+    public function setDomains(array $domains): void
+    {
+        $this->domains = $domains;
+    }
+
+    /**
+     * @return \Generator<MessageId, array{domain: Domain, locale: Locale, message: string}, void, void>
+     */
+    private function getTranslations(MessageCatalogueInterface ...$catalogues): iterable
+    {
+        foreach ($catalogues as $catalogue) {
+            $locale = $catalogue->getLocale();
+            foreach ($catalogue->getDomains() as $domain) {
+                if (\in_array($domain, $this->excludedDomains, true)) {
+                    continue;
+                }
+                if (0 !== \count($this->domains) && !\in_array($domain, $this->domains, true)) {
+                    continue;
+                }
```

Comment:
I think we maybe should use "includedDomain" / "excludedDomain" or something similar, maybe just me but the difference in naming makes it less clear